smoketests/tests/fail_initial_publish.py

```diff
@@ -0,0 +1,75 @@
+from .. import Smoketest, random_string
+import subprocess
+
+class FailInitialPublish(Smoketest):
```

Comment:
can you also add a test for the case where the initial publish is successful but subsequent publish fails

Reply:
We have that case already: lines 64 through 75 of this file publish the broken module on top of the working one, and then check that the working module still works.

Reply:
it's _slightly_ different: we publish a broken module, then a working one, then a broken one. We don't test where the first publish was working. I mainly ask from a "black box" POV since that's one of the cases that I tested with when reproducing it manually. It's probably not a meaningful distinction though :shrug: